util/csvtojson.py

```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env python
+
+import json, sys, csv
+
+def convert_flat_type_func(column_type):
+  column_type = column_type.lower()
+  if column_type == "integer" or column_type == "int" or column_type == "bigint":
+    return lambda x: None if x == "" else int(x)
+  elif column_type == "float" or column_type == "numeric":
+    return lambda x: None if x == "" else float(x)
+  elif column_type == "text":
+    return lambda x: None if x == "" else x
+  elif column_type == "boolean":
+    return lambda x: None if x == "" else x == "t"
+  else:
+    raise ValueError("Unsupported data type %s" %column_type)
+
+# given a column type, returns a function that takes a string input
+# and output with the correct type
+def convert_type_func(column_type):
+  column_type = column_type.lower()
+  if column_type.endswith("[]"):
+    # csv array starts and ends with curly braces
+    column_type = column_type[:-2]
+    # string unescaping
+    # note this is an array as a csv field, we first unescape csv escaping,
+    # and then do array unescaping, apply csv escaping, and parse the arry
+    # elements as csv
+    # csv escapes double quote with two double quotes
+    # array escapes double quote using backslashes
+    flat_func = convert_flat_type_func(column_type)
+    def convert_text_array_func(value):
+      if value == "": return None
+      arr = csv.reader([value[1:-1].replace('""', '"').replace('\\"', '""')], delimiter=',', quotechar='"').next()
```

Comment:
Here, the array is inside the csv file, so the array itself is double quoted. Every double quote inside the array is doubled. After we unescape the csv and psql array, the elements of array become normal csv escaped elements.